finat/tensor_product.py

```diff
@@ -29,7 +29,35 @@ def cell(self):
 
     @property
     def degree(self):
-        raise NotImplementedError("Unused property.")
+        return tuple(fe.degree for fe in self.factors)
+
+    @cached_property
+    def _entity_dofs(self):
+        shape = tuple(fe.space_dimension() for fe in self.factors)
+        entity_dofs = {}
+        for dim in product(*[fe.cell.get_topology().keys()
+                             for fe in self.factors]):
+            entity_dofs[dim] = {}
+            topds = [fe.entity_dofs()[d]
+                     for fe, d in zip(self.factors, dim)]
+            for tuple_ei in product(*[sorted(topd) for topd in topds]):
+                tuple_vs = list(product(*[topd[ei]
+                                          for topd, ei in zip(topds, tuple_ei)]))
+                if tuple_vs:
+                    vs = list(numpy.ravel_multi_index(numpy.transpose(tuple_vs), shape))
+                else:
+                    vs = []
+                entity_dofs[dim][tuple_ei] = vs
+            # flatten entity numbers
+            entity_dofs[dim] = dict(enumerate(entity_dofs[dim][key]
```

Comment:
And then overwrite here.  Maybe it would be clearer if these were two different variables?